src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -1242,5 +1220,74 @@ public static void ConvertFromBlobToNumbers(JsonOperationContext context, Counte
             }
         }
 
+
+        private static ConflictStatus CompareCounterValuesAndDeletedCounter(BlittableJsonReaderObject.RawBlob counterValues, string deletedCounter,
+            BlittableJsonReaderArray dbIds, bool remoteDelete)
+        {
+            //any missing entries from a change vector are assumed to have zero value
+            var blobHasLargerEntries = false;
+            var deleteHasLargerEntries = false;
+
+            var deletedCv = deletedCounter.ToChangeVector();
+
+            var sizeOfValues = counterValues.Length / SizeOfCounterValues;
+
+            int numOfMatches = 0;
+            for (int i = 0; i < sizeOfValues; i++)
+            {
+                bool found = false;
+
+                var current = (CounterValues*)counterValues.Ptr + i;
+                var dbId = dbIds[i].ToString().Substring(2);
```

Comment:
let's find a way to do that without materialization

Reply:
done